core/chains/evm/client/errors.go

```diff
@@ -452,7 +452,16 @@ func (err JsonError) Error() string {
 	return err.Message
 }
 
-func (err *JsonError) String() string {
+// Satisfies rpc.Error interface, which any actual jsonError returned from geth is expected to
```

Comment:
Do we also need to satisfy `rpc.DataError`? `ErrorData()` seems redundant.
Should we add interface guard to ensure JsonError always implements rpc.Error?

Reply:
There's no present need for it, but I guess it probably makes sense to add it so that it's fully parallel with the `jsonError` type defined in `go-ethereum`

Reply:
Oh, I'm glad you asked this--I was thinking I hadn't added `ErrorData()`, but looks like I'd started to add it but hadn't actually finished it. Just completed that, and added a type guard for both.

The jsonError's returned from geth satisfy both rpc.Error and rpc.DataError--presently we only use rpc.Error but having both could avoid a future mistake where we're expecting it to satisfy that interface and it doesn't.

Reply:
For reference:
https://github.com/ethereum/go-ethereum/blob/master/rpc/json.go#L139-L158